### PR TITLE
removed array in-place deprecated warning,  decorators.  updated test.

### PR DESCRIPTION
**Pull request by lfuhrman**
update for TODO for version 2.0: 

* In ``skimage/morphology/misc.py`` remove the deprecated parameter
  ``in_place`` in remove_small_holes(), remove_small_objects() and
  clear_border(), and update the tests.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

**Comment by grlee77** [comment]
please also remove from TODO.txt

**Comment by grlee77** [comment]
This item was originally in the TODO for 1.0, but SKIP 4 has since proposed not introducing breaking changes in 1.0. Given that, this has been tagged for `skimage2`.

File: skimage/morphology/misc.py [hunk 1]
```diff
@@ -47,8 +47,6 @@ def _check_dtype_supported(ar):
                         "Got %s." % ar.dtype)
 
 
-@remove_arg("in_place", changed_version="1.0",
-            help_msg="Please use out argument instead.")
 def remove_small_objects(ar, min_size=64, connectivity=1, in_place=False,
```
Comment (grlee77) [inline skimage/morphology/misc.py:50]:
```suggestion
def remove_small_objects(ar, min_size=64, connectivity=1, 
```

also need to remove the original argument :)

**Review by grlee77 — APPROVED** [review]
Thanks @lfuhrman, this looks good. Once we branch 1.0 we will be able to merge this into main.

**Comment by lagru** [comment]
I think this was recently addressed and merged in [`a91f4f8`](https://github.com/scikit-image/scikit-image/commit/a91f4f8bab92c4eede48e4949805c748f436c4de#diff-44c001454a4cbcff189fa124ece126542e1aa9b120ae9666a178b23d5b356805) so I'm gonna close this. Please feel free to request or reopen if that's not correct.

@lfuhrman, sorry for missing this PR! We had some discussion on when to use version 1.0 and decided to do a 0.20 release before. I guess while renaming the milestones and stuff this got pushed to the wrong milestone and left our radar.

**Pull request closed without merging**